src/Management/src/Endpoint/Actuators/CloudFoundry/PermissionsProvider.cs

```diff
@@ -75,23 +75,34 @@ public async Task<SecurityResult> GetPermissionsAsync(string accessToken, Cancel
                 _logger.LogInformation("Cloud Foundry returned status: {HttpStatus} while obtaining permissions from: {PermissionsUri}", response.StatusCode,
                     checkPermissionsUri);
 
-                return response.StatusCode is HttpStatusCode.Forbidden or HttpStatusCode.Unauthorized
-                    ? new SecurityResult(HttpStatusCode.Forbidden, AccessDeniedMessage)
+                if (response.StatusCode is HttpStatusCode.Forbidden)
+                {
+                    return new SecurityResult(HttpStatusCode.Forbidden, AccessDeniedMessage);
+                }
+
+                return (int)response.StatusCode is > 399 and < 500
+                    ? new SecurityResult(HttpStatusCode.Unauthorized, InvalidTokenMessage)
                     : new SecurityResult(HttpStatusCode.ServiceUnavailable, CloudfoundryNotReachableMessage);
             }
 
-            EndpointPermissions permissions = await GetPermissionsAsync(response, cancellationToken);
+            EndpointPermissions permissions = await ParsePermissionsAsync(response, cancellationToken);
             return new SecurityResult(permissions);
         }
         catch (Exception exception) when (!exception.IsCancellation())
         {
             _logger.LogError(exception, "Cloud Foundry returned exception while obtaining permissions from: {PermissionsUri}", checkPermissionsUri);
 
+            return new SecurityResult(HttpStatusCode.InternalServerError, exception.Message);
+        }
+        catch (Exception exception)
+        {
+            _logger.LogInformation(exception, "Task cancelled or timed out while obtaining permissions from: {PermissionsUri}", checkPermissionsUri);
+
```

Comment:
This catch-all should be removed, so that the cancellation bubbles up and terminates the entire pipeline.

While it's useful to log a timeout, a client walking away because they closed their browser, or a consuming machine that restarts, should not pollute log files, as this is normal and expected behavior.

I would expect a true timeout while connecting to the auth server to bubble up and be caught and logged by ASP.NET itself, similar to any other exception type.